doc/high-level.md

```diff
@@ -0,0 +1,83 @@
+# High Level Bindings
+
+The purpose of the high-level magicl bindings is to allow for MATLAB-like multidimensional arrays in lisp. 
+
+## Constructors
+
+The construction of tensors can be done with any of the given constructors. The constructors take a shape and the arguments for method of construction.
+
+Tensors are specialized on both the shape and the element type. The class of a tensor will be of the form `$CLASS/$TYPE`. (eg. `MATRIX/DOUBLE-FLOAT`)
+
+| Number of dimensions | Tensor Class    |
+|----------------------|-----------------|
+| 1                    | `VECTOR`        |
+| 2                    | `MATRIX`        |
+| *                    | `TENSOR`        |
+
+| Element Type             | Class Suffix           |
+|--------------------------|------------------------|
+| `SINGLE-FLOAT`           | `SINGLE-FLOAT`         |
+| `DOUBLE-FLOAT`           | `DOUBLE-FLOAT`         |
+| `(COMPLEX SINGLE-FLOAT)` | `COMPLEX-SINGLE-FLOAT` |
+| `(COMPLEX DOUBLE-FLOAT)` | `COMPLEX-DOUBLE-FLOAT` |
+| `(SIGNED-BYTE 32)`       | `INT32`                |
+
+The type of the elements of the tensor can be specified with the `:type` keyword, or the constructor will attempt to find an appropriate type from the given arguments. The default element type for a tensor is `DOUBLE-FLOAT`.
+
+The layout of the tensor (column-major or row-major) can be specified with the `:layout` keyword. This affects the underlying storage of the tensor and will affect how it carries out operations with LAPACK. 
+
+
+## Other Library Equivalents
+
+This table was adapted largely from the [NumPy Equivalent Table](https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html#linear-algebra-equivalents).
+
+### Basic Accessors
+
+| MAGICL         | MATLAB     | NumPy                   | Description                                                   |
+|----------------|------------|-------------------------|---------------------------------------------------------------|
+| `(layout a)`    | `ndims(a)` | `ndim(a)` or `a.ndim`   | Get the number of dimensions of the array.                    |
+| `(size a)`     | `numel(a)` | `size(a)` or `a.size`   | Get the number of elements of the array.                      |
+| `(shape a)`    | `size(a)`  | `shape(a)` or `a.shape` | Get the shape of the array.                                   |
+| `(tref a 1 4)` | `a(2,5)`   | `a[1, 4]`               | Get the element in the second row, fifth column of the array. |
```

Comment:
For vectors and tensors, `tref` seems to do something even for nonsensical indices. For example, both
`(tref (from-list '(1d0 2d0 3d0 4d0) '(4)) 1 0 0 1)` and 
`(tref (from-list '(1d0 2d0 3d0 4d0) '(1 2 2)) 1 0 0 1)` give `2.0d0`,
but `(tref (from-list '(1d0 2d0 3d0 4d0) '(2 2)) 1 0 0 1)` fails an assertion.

Reply:
Looks like vectors and tensors are missing a `valid-matrix-p`. Will fix.

Reply:
Fixed in 4b17a3b